drivers/bluetooth/hci/rpmsg.c

```diff
@@ -317,9 +325,55 @@ static int bt_rpmsg_open(void)
 	return 0;
 }
 
+static int bt_rpmsg_close(void)
+{
+	int err;
+	struct net_buf *buf = NULL;
+	struct net_buf *rsp = NULL;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_RESET, 0);
+	if (!buf) {
+		LOG_ERR("Allocating command buffer failed!");
+		return -ENOBUFS;
+	}
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_RESET, NULL, &rsp);
+	if (err) {
+		LOG_ERR("Sending reset command failed with: %d", err);
+		return err;
+	}
```

Comment:
Does the other drivers also send an HCI reset on close?

Reply:
> Is there a reason why we, for just this driver, are sending an HCI reset when closing the driver? Should we not do it here? Should we do it for the other drivers?
> 
> Just seems odd to have this change in behavior from driver to driver

I agree. We could potentially also control this using driver-specific quirk flags, so that the host-side is responsible for doing the reset if the driver claims that it needs it.

Reply:
@Thalley @jhedberg This implementation was based on the ST one. I don't know what ST's radio is doing on custom reset. For me sending an HCI reset before power cycling makes this operation less brutforce.